cmd/server-main.go

```diff
@@ -369,6 +369,11 @@ func serverMain(c *cli.Context) {
 	// Check if requested port is available.
 	host, portStr, err := net.SplitHostPort(serverAddr)
 	fatalIf(err, "Unable to parse %s.", serverAddr)
+	if portStr == "0" {
+		// Zero port means choose any available port, avoids this since it won't work
+		// with the distributed config and all clients will be confused too.
+		fatalIf(errInvalidArgument, "Cannot accept zero port number")
```

Comment:
``fatalIf(errInvalidArgument, `Invalid port %s, please use `--address` to pick a specific port number`)``